packages/@aws-cdk/aws-dynamodb/lib/table-time-to-live-handler/index.ts

```diff
@@ -0,0 +1,72 @@
+import { DynamoDB } from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+import { disableTimeToLive, enableTimeToLive, tableWillBeRemoved, timeToLiveStatus } from './utils';
+
+const dynamodb = new DynamoDB({ apiVersion: '2012-08-10' });
+
+export async function onEventHandler(event: OnEventRequest): Promise<OnEventResponse> {
+  console.log('Event: %j', event);
+
+  if (event.RequestType === 'Create' || event.RequestType === 'Update') {
+    const currentTtl = await dynamodb.describeTimeToLive({
+      TableName: event.ResourceProperties.TableName,
+    }).promise();
+    console.log('Describe time to live: %j', currentTtl);
+
+    switch (currentTtl.TimeToLiveDescription?.TimeToLiveStatus) {
+      case 'ENABLING':
+        throw new Error('Can not change time to live while it is being enabled.');
+      case 'DISABLING':
+        throw new Error('Can not change time to live while it is being disabled.');
+      case 'ENABLED':
+        // If the current TTL is different from the requested TTL the first step is always to disable the TTL.
+        // Should the TTL not be undefined it will be enabled with the new attribute as this function will be
+        // called again and the 'DISABLED' case will enable it.
+        if (event.ResourceProperties.TimeToLiveAttribute !== currentTtl.TimeToLiveDescription.AttributeName) {
+          await disableTimeToLive(event);
+        }
+        break;
+      case 'DISABLED':
+        if (event.ResourceProperties.TimeToLiveAttribute) {
+          await enableTimeToLive(event);
+        }
+        break;
+    }
+
+    return { PhysicalResourceId: event.ResourceProperties.TableName };
+  }
+
+  if (! await tableWillBeRemoved(event)) {
+    await disableTimeToLive(event);
+  }
+
+  return {};
+}
+
+export async function isCompleteHandler(event: IsCompleteRequest): Promise<IsCompleteResponse> {
+  console.log('Event: %j', event);
+
+  let { stable, correct } = await timeToLiveStatus(event);
+
+  if (stable && !correct) {
+    // If the ttl is stable but incorrect, we are in the second step of the enabled -> disabled -> enabled flow.
+    // We therefore call the onEventHandler as the logic is the same as going from disabled to enabled,
+    // which the handler supports. Afterwards we need to get the current ttl status as it should have changed.
+    // This makes it possible to change the ttl attribute from a to b.
+    await onEventHandler(event);
+    ({ stable, correct } = await timeToLiveStatus(event));
+  }
+
+  switch (event.RequestType) {
+    case 'Create':
+    case 'Update':
+    case 'Delete':
+      if (event.RequestType === 'Delete' && await tableWillBeRemoved(event)) {
+        // There is nothing to delete, as time to live is connected to the table
+        return { IsComplete: true };
+      }
+      // Complete when time to live is fully functional
+      return { IsComplete: stable && correct };
```

Comment:
We always repeat `stable && correct` here.

Let's just change `timeToLiveStatus()` to return a single `boolean`. If that's the case, a different name is probably in order (perhaps `isTimeToLiveStable()`?).

Reply:
Ah, I see I'm wrong here (`stable && !correct` and `stable && correct`)

Can you switch to `timeToLiveStatus()` to return an enum instead? I have a feeling other people will make the same mistake I did here 🙂.

Reply:
Sounds good, I have changed it to an enum representing the four possible states. Do you think the naming is ok or do you have further suggestions to improve clarity?

Reply:
Looks good! If it were me, I would make 2 changes:

1. I would define the `TimeToLiveStatus` right above where it's first used (the `getTimeToLiveStatus()` function).
2. I would make it have 3 values (`UNSTABLE`, `STABLE_INCORRECT` and `STABLE_CORRECT`), as you never use the `INSTABLE_AND_CORRECT` and `INSTABLE_AND_INCORRECT` values.

But these are relatively minor.